ViewModels/UserBoardViewModel.cs

```diff
@@ -0,0 +1,128 @@
+using Combat_Critters_2._0.Services;
+using CombatCrittersSharp.exception;
+using CombatCrittersSharp.objects.user;
+
+namespace Combat_Critters_2._0.ViewModels
+{
+    public class UserBoardViewModel : INotifyPropertyChanged
+    {
+        private BackendService _backendService;
+        private ObservableCollection<IUser> _allUsers;
+        private bool _hasUsers;
+        private ObservableCollection<IUser> _filteredUser;
+
+        public bool HasUsers
+        {
+            get => _hasUsers;
+            set
+            {
+                _hasUsers = value;
+                OnPropertyChanged(nameof(HasUsers));
+            }
+        }
+        public ObservableCollection<IUser> AllUsers
+        {
+            get => _allUsers;
+            set
+            {
+                _allUsers = value;
+                OnPropertyChanged(nameof(AllUsers));
+            }
+        }
+
+        public ObservableCollection<IUser> FilteredUser
+        {
+            get => _filteredUser;
+            set
+            {
+                _filteredUser = value;
+                OnPropertyChanged(nameof(FilteredUser));
+            }
+        }
+
+        public UserBoardViewModel()
+        {
+            _backendService = new BackendService(ClientSingleton.GetInstance("http://api.combatcritters.ca:4000"));
+            _allUsers = new ObservableCollection<IUser>();
+            _filteredUser = new ObservableCollection<IUser>();
+            HasUsers = false;
+            //start Loading the user cards.
+            Task.Run(async () => await InitializeViewModelAsync());
+        }
+
+        private async Task InitializeViewModelAsync()
+        {
+            await LoadUsers();
+        }
+
+        /// <summary>
+        /// Load all game users 
+        /// </summary>
+        /// <returns></returns>
+        private async Task LoadUsers()
+        {
+            try
+            {
+                var users = await _backendService.GetUsersAsync(); //get all users from backendf
+
+                if (users != null && users.Count > 0)
+                {
+                    Application.Current?.Dispatcher.Dispatch(() =>
+                    {
+                        AllUsers = new ObservableCollection<IUser>(users);
+                        HasUsers = true;
+                    });
+                }
+                else
+                {
+                    //Game has no Users
+                    HasUsers = false;
+                    AllUsers.Clear();
+                }
+
+            }
+            catch (RestException)
+            {
+                HasUsers = false;
+
+                if (Application.Current?.MainPage != null)
+                    await Application.Current.MainPage.DisplayAlert("Error", "Failed to load users. Please try again.", "OK");
+            }
+            catch (Exception)
+            {
+                HasUsers = false;
+                throw; //bubble up to the global exception
+            }
+
```

Comment:
exception handlers probably shouldn't change instance state, cleaner way to do this is a function scoped bool that is initialized false and set `HasUsers` to the value of that bool at the end of the function

Reply:
Ah okay